File: 107_limitswitch/src/RaspberryPI/limitswitch.py
# coding: utf-8
#
# FaBo Brick Sample
#
# brick_analog_limitswitch
#

import RPi.GPIO as GPIO
import time

LEDPIN = 4 
LSPIN = 7   #LimitSwitch pin

led_state = 0	

GPIO.setwarnings(False)
GPIO.setmode( GPIO.BCM )
GPIO.setup( LEDPIN, GPIO.OUT )
GPIO.setup( LSPIN, GPIO.IN)

while True:
    if( GPIO.input( LSPIN ) ):
         led_state = 1 - led_state
    GPIO.output( LEDPIN, led_state )
    print "led_state: %d " % led_state
    time.sleep(0.2)

File: 0107_limitswitch/src/Edison/limitswitch.js
//
// FaBo Brick Sample
//
// #107 LimitSwitch Brick
//

//library
var m = require('mraa');

//pin setup
var myButton = new m.Gpio(14); //Button A0
var myLed    = new m.Gpio(2);  //LED D2

myButton.dir(m.DIR_IN);     //Button input
myLed.dir(m.DIR_OUT);       //LED output

//call loop function
loop();


function loop()
{

  if (myButton.read()){
    myLed.write(1);
  }
  else {
    myLed.write(0);
  }

  setTimeout(loop,100);
}
